src/NzbDrone.Core/Notifications/Slack/SlackService.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Core.Notifications.Slack.Payloads;
+using NzbDrone.Core.Organizer;
+using NzbDrone.Core.Rest;
+using NzbDrone.Core.Tv;
+using NzbDrone.Core.Validation;
+using RestSharp;
+
+namespace NzbDrone.Core.Notifications.Slack
+{
+    public interface ISlackService
+    {
+        void OnDownload(DownloadMessage message, SlackSettings settings);
+        void OnRename(Series series, SlackSettings settings);
+        void OnGrab(GrabMessage message, SlackSettings settings);
+        ValidationFailure Test(SlackSettings settings);
+    }
+
+    public class SlackService : ISlackService
+    {
+        private readonly Logger _logger;
+
+        public SlackService(Logger logger)
+        {
+            _logger = logger;
+        }
+
+        public void OnDownload(DownloadMessage message, SlackSettings settings)
+        {
+            var payload = new SlackPayload()
+            {
+                IconEmoji = settings.Icon,
+                Username = settings.BotName,
+                Text = "Downloaded",
+                Attachments = new List<Attachment>()
+                {
+                    new Attachment()
+                    {
+                        Fallback = message.Message,
+                        Title = message.Series.Title,
+                        TitleLink = $"http://www.imdb.com/title/{message.Series.ImdbId}/",
+                        Text = message.Message,
+                        Color = "good"
```

Comment:
Is this a magic colour that Slack allows? Any reason why we don't use it for other Slack notifications?